core/indexer/elasticIndexer.go

```diff
@@ -0,0 +1,461 @@
+package indexer
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"io"
+
+	"github.com/ElrondNetwork/elrond-go/core/statistics"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+)
+
+type elasticIndexer struct {
+	*txDatabaseProcessor
+
+	elasticClient databaseClientHandler
+	parser        *dataParser
+}
+
+// NewElasticIndexer creates an elasticsearch es and handles saving
+func NewElasticIndexer(arguments ElasticIndexerArgs) (ElasticIndexer, error) {
+	ei := &elasticIndexer{
+		elasticClient: arguments.DBClient,
+		parser: &dataParser{
+			hasher:      arguments.Hasher,
+			marshalizer: arguments.Marshalizer,
+		},
+	}
+
+	ei.txDatabaseProcessor = newTxDatabaseProcessor(
+		arguments.Hasher,
+		arguments.Marshalizer,
+		arguments.AddressPubkeyConverter,
+		arguments.ValidatorPubkeyConverter,
+	)
+
+	err := ei.init(arguments.IndexTemplates, arguments.IndexPolicies)
+	if err != nil {
+		return nil, err
+	}
+
+	return ei, nil
+}
+
+func (ei *elasticIndexer) init(indexTemplates, indexPolicies map[string]io.Reader) error {
+	err := ei.createOpenDistroTemplates(indexTemplates)
+	if err != nil {
+		return err
+	}
+
+	err = ei.createIndexPolicies(indexPolicies)
+	if err != nil {
+		return err
+	}
+
+	err = ei.createIndexTemplates(indexTemplates)
+	if err != nil {
+		return err
+	}
+
+	err = ei.createIndexes()
+	if err != nil {
+		return err
+	}
+
+	err = ei.setInitialAliases()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ei *elasticIndexer) createIndexPolicies(indexPolicies map[string]io.Reader) error {
+	txp := getTemplateByName(txPolicy, indexPolicies)
+	if txp != nil {
+		err := ei.elasticClient.CheckAndCreatePolicy(txPolicy, txp)
+		if err != nil {
+			return err
+		}
+	}
+
+	blockp := getTemplateByName(blockPolicy, indexPolicies)
+	if blockp != nil {
+		err := ei.elasticClient.CheckAndCreatePolicy(blockPolicy, blockp)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ei *elasticIndexer) createOpenDistroTemplates(indexTemplates map[string]io.Reader) error {
+	opendistroTemplate := getTemplateByName("opendistro", indexTemplates)
+	if opendistroTemplate != nil {
+		err := ei.elasticClient.CheckAndCreateTemplate("opendistro", opendistroTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ei *elasticIndexer) createIndexTemplates(indexTemplates map[string]io.Reader) error {
+	txTemplate := getTemplateByName(txIndex, indexTemplates)
```

Comment:
could have iterate over a slice of strings. Could have had a variable called
```GO
var availableIndexes = []string{blockIndex, miniblocksIndex, txIndex...........}
```

and called them in a for range

Reply:
refactored